server/middlewares/validations/ReportValidation.js

```diff
@@ -51,6 +54,119 @@ class ReportValidation {
     });
   }
 
+  /**
+   * @description validate report input
+   *
+   * @param {object} req
+   * @param {object} res
+   * @param {func} next
+   *
+   * @return {void}
+   */
+  static validateReportCategory(req, res, next) {
+    const { title, description } = req.body;
+    const data = {
+      title,
+      description,
+    };
+
+    const rules = {
+      title: 'required|max:1000|min:5',
+      description: 'required|max:1000|min:10',
+    };
+
+    const message = {
+      'required.title': 'The :attribute field is required.',
+      'max.title': 'The :attribute is too long. Max length is :max characters.',
+      'min.title': 'The :attribute is too short. Min length is :min characters.',
+      'required.description': 'The :attribute field is required.',
+      'max.description': 'The :attribute is too long. Max length is :max characters.',
+      'min.description': 'The :attribute is too short. Min length is :min characters.',
+    };
+
+    const validation = new Validator(data, rules, message);
+
+    if (validation.passes()) {
+      return ReportCategory.findOne({
```

Comment:
I am a little confused as to why this check is being done in here, could you please shed a little light on it, thank you.

Reply:
I believe validation involves checking the validity and existence of the user's input, that is why I checked here, also, to avoid unnecessarily long create/update functions.